src/main/scala/it/pagopa/interop/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -121,6 +121,76 @@ class PartyApiServiceImpl(
 
   }
 
+  /** Code: 200, Message: successful operation, DataType: Institution
+   * Code: 405, Message: Institution not found, DataType: Problem
+   * Code: 400, Message: Invalid ID supplied, DataType: Problem
+   */
+  override def updateInstitutionById(id: String, institution: Institution)(implicit
+    toEntityMarshallerInstitution: ToEntityMarshaller[Institution],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Updating institution $id")
+
+    val commander: EntityRef[Command] = getCommander(id)
+      .map(shard => sharding.entityRefFor(PartyPersistentBehavior.TypeKey, shard.toString))
+      .toList
+
+    val updatedInstitution: Future[StatusReply[Party]] = for {
+      uuid      <- id.toFutureUUID
+      found <- commander.ask(ref => GetParty(uuid, ref))
+      party       <- found.toFuture(InstitutionNotFound(id))
+      updatedOrg <- maybeExistingOrg
+        .map(i =>
+          InstitutionParty(
```

Comment:
I suggest here to be more strict on which field you can update.
For example, in my opinion, `externalId` must not change.

Reply:
Now we are checking if this fied is changing, blocking the operation